### PR TITLE
[GridBuilder] Predefined filters

### PR DESCRIPTION
**Pull request by loic425**
Related to #155 

```php
<?php
// config/sylius/grids/book.php

use App\Entity\Author;
use App\Entity\Book;
use App\BookStatus;
use Sylius\Bundle\GridBundle\Builder\GridBuilder;
use Sylius\Bundle\GridBundle\Builder\Filter\SelectFilter;
use Sylius\Bundle\GridBundle\Builder\Filter\StringFilter;
use Sylius\Bundle\GridBundle\Config\GridConfig;

return static function (GridConfig $grid) {
    $grid->addGrid(GridBuilder::create('app_book', Book::class)
        ->addFilter(StringFilter::create('search', ['name', 'author.firstName', 'author.lastName']))
        ->addFilter(SelectFilter::create('status', BookStatus::CHOICES))
        ->addFilter(EntityFilter::create('author', Author::class))
    );
};
```

### CODE REVIEW

File: src/Bundle/Tests/DependencyInjection/GridBuilderConfigurationTest.php [hunk 1]
```diff
@@ -137,6 +145,114 @@ public function it_builds_grid_with_filters(): void
...
+    /**
+     * @test
+     */
+    public function it_builds_grid_with_predefined_filters(): void
+    {
+        $gridBuilder = GridBuilder::create('app_admin_book', Book::class)
+            ->addFilter(StringFilter::create('search'))
+            ->addFilter(DateFilter::create('createdAt'))
+            ->addFilter(BooleanFilter::create('enabled'))
+            ->addFilter(EntityFilter::create('author', Author::class))
```
[the remark this answers is not in the document]
Comment (loic425) [reply src/Bundle/Tests/DependencyInjection/GridBuilderConfigurationTest.php:190]: I did but it proves nothing cause my parameter is not converted on my configuration test. It just prove you can set any string you want.

[the remark this answers is not in the document]
Comment (loic425) [reply src/Bundle/Tests/DependencyInjection/GridBuilderConfigurationTest.php:190]: https://github.com/Sylius/SyliusGridBundle/pull/175/commits/79e60cb699df6c4e591306f7c87e6146a4386678

File: src/Bundle/spec/Builder/Filter/MoneyFilterSpec.php [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+use Sylius\Bundle\GridBundle\Builder\Filter\MoneyFilter;
+
+final class MoneyFilterSpec extends ObjectBehavior
+{
+    function it_is_initializable(): void
+    {
+        $this->shouldHaveType(MoneyFilter::class);
+    }
+
+    function it_creates_money_filters(): void
```
[the remark this answers is not in the document]
Comment (loic425) [reply src/Bundle/spec/Builder/Filter/MoneyFilterSpec.php:18]: done

**Review by lchrusciel — APPROVED** [review]

File: src/Bundle/Tests/DependencyInjection/GridBuilderConfigurationTest.php (hunk 1 above)
[the remark this answers is not in the document]
Comment (lchrusciel) [reply src/Bundle/Tests/DependencyInjection/GridBuilderConfigurationTest.php:187]: That makes this feature a little bit less usable for Sylius, as we cannot be sure about end class. Still, it is nice for endusers

**Comment by lchrusciel** [comment]
Thank you, Loïc! :tada:

File: src/Bundle/Tests/DependencyInjection/GridBuilderConfigurationTest.php (hunk 1 above)
[the remark this answers is not in the document]
Comment (loic425) [reply src/Bundle/Tests/DependencyInjection/GridBuilderConfigurationTest.php:187]: You can use any parameter, so this can be used on Sylius too.

**Pull request merged**